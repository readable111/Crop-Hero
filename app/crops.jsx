/****
 * @author Isaac Boodt
 * @reviewer Daniel Moreno
 * @tester 
 ***/

import { StatusBar } from 'expo-status-bar';
import React from 'react';
import { Pressable, StyleSheet, Text, View } from 'react-native';
import { useRouter } from 'expo-router'
import NavBar from '../assets/NavBar.jsx'
import Colors from '../assets/Color';
import { useFonts } from 'expo-font';
const crops = () => {
  const router = useRouter();
  const [fontsLoaded, fontError] = useFonts({
    'WorkSans-Semibold': require('../assets/fonts/WorkSans-SemiBold.ttf'),
    'Domine-Medium': require('../assets/fonts/Domine-Medium.ttf'),
    'Domine-Regular': require('../assets/fonts/Domine-Regular.ttf'),
});
  return (
    <View style={{
      flex: 1,
    }}>
      <Text style={styles.title}>My Crops</Text>
      {/*<View style={styles.circle}></View>*/} 
      
      <View style={styles.background}>
        <View style={styles.button}>
          <Pressable onPress={() => router.push("/addcrops")}>
            <Text style={styles.buttonText}>Add Crop</Text>
          </Pressable>
        </View>
        <View style={styles.button}>
          <Pressable onPress = {() => router.push("/viewcrops")}>
            <Text style={styles.buttonText}>View Crop</Text>
          </Pressable>
        </View>
      <NavBar cropsSelected/>
      </View>
    </View>

  );
}

export default crops;

const styles = StyleSheet.create({
  container: {
    flex: 4,
    backgroundColor: Colors.SANTA_GRAY,
  },
  circleBlock:{
    backgroundColor: 'white',
    position: 'absolute',
    marginTop: -20,

  },
  title:{
    flex: .3,
    backgroundColor: Colors.ALMOND_TAN,
    borderColor: '#20232a',
    borderWidth: 1,
    padding: 10,
    textAlign: 'right',
    alignContent: 'center',
    fontSize: 42,
    fontFamily: 'Domine-Medium'
  },
  backdrop:{
    flex: 10,
    backgroundColor: '#FFFFFF',
    marginTop: 0,
    position: 'absolute',
  },
  background:{
    flex: 3,
    marginBottom: 'auto',
    backgroundColor: Colors.SANTA_GRAY,
  },
  circle:{
    position: 'absolute',
    marginTop: -50,
    marginLeft: 20,
    width: 100,
    height: 110,
    borderRadius: 100 / 2,
    backgroundColor: "lime",
    overflow: 'hidden',
  },
  button:{
    backgroundColor: Colors.SCOTCH_MIST_TAN,
    textAlign: 'center',
    padding: 30,
    fontSize: 38,
    marginTop: 60,
    marginHorizontal: 20,
    borderColor: '#20232a',
    borderWidth: 1,
    borderRadius: 8,
  },
  buttonText:{
    textAlign: 'center',
    fontSize: 32,
    fontFamily: 'Domine-Regular'
  },
  whitebox: {
    backgroundColor: 'white',
    marginTop: -40
  },
  pressable: {
    marginTop: 70
  }
});

