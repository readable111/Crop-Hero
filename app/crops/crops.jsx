/****
 * @author Isaac Boodt
 * @reviewer Daniel Moreno
 * @tester 
 ***/

import { StatusBar } from 'expo-status-bar';
import { React, useEffect, useState} from 'react';
import { Pressable, StyleSheet, Text, View } from 'react-native';
import { useRouter } from 'expo-router'
import NavBar from '../../assets/NavBar.jsx'
import Colors from '../../assets/Color.js';
import AsyncStorage from '@react-native-async-storage/async-storage';
import { useFonts } from 'expo-font';
const crops = () => {


  const router = useRouter();
  const [fontsLoaded, fontError] = useFonts({
    'WorkSans-Semibold': require('../../assets/fonts/WorkSans-SemiBold.ttf'),
    'Domine-Medium': require('../../assets/fonts/Domine-Medium.ttf'),
    'Domine-Regular': require('../../assets/fonts/Domine-Regular.ttf'),
});
const [isDark, setIsDarkMode] = useState(false)
        useEffect(() => {
                const fetchDarkModeSetting = async () => {
                        const JSON_VALUE = await AsyncStorage.getItem('dark_mode_setting');
                        let result = null
                        if(JSON_VALUE && JSON_VALUE !== "")
                        {
                                result = JSON.parse(JSON_VALUE)
                                //console.log("Async: " + result)
                        }
                        else
                        {
                                useColorScheme.Appearence.getColorScheme()
                                if(colorScheme == 'dark')
                                {
                                        result = true;
                                }
                                else
                                {
                                        result = false;
                                }
                                console.log("colorScheme: " + result)
                        }
                        setIsDarkMode(result)
                }
                fetchDarkModeSetting()
                .catch(console.error);
        }, [])
  return (
    <View style={[ styles.container, isDark && styles.containerDark]}>
      <View style={[styles.titleCard, isDark && styles.titleCarddark]}>
        <Text style={[styles.title, isDark && {color: Colors.WHITE_SMOKE}]}>My Crops</Text>
      </View>
      {/*<View style={styles.circle}></View>*/} 
        <View style={ [ styles.button, isDark && styles.buttonDark ] }>
          <Pressable testID = "addCrops" onPress={() => router.push("./addcrops", {relativeTopDirectory: true})}>
            <Text style={styles.buttonText}>Add Crop</Text>
          </Pressable>
        </View>
        <View style={[ styles.button, isDark && styles.buttonDark ]}>
          <Pressable onPress = {() => router.push("./viewcrops", {relativeToDirectory: true})}>
            <Text style={styles.buttonText}>View Crop</Text>
          </Pressable>
        </View>
      <NavBar cropsSelected darkMode={isDark}/>
    </View>

  );
}

export default crops;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: Colors.SANTA_GRAY,
  },
  containerDark:{
    backgroundColor: Colors.BALTIC_SEA
},
  circleBlock:{
    backgroundColor: 'white',
    position: 'absolute',
    marginTop: -20,

  },
  titleCard:{
    backgroundColor: Colors.ALMOND_TAN,
    borderColor: Colors.CHARCOAL,
    borderWidth: 1,
    padding: 18,
  },
  titleCarddark:{
    backgroundColor: Colors.CHARCOAL
  },
  title:{
    textAlign: 'right',
    fontSize: 42,
    fontFamily: 'Domine-Medium',
    alignContent: 'center'
  },
  backdrop:{
    flex: 10,
    backgroundColor: '#FFFFFF',
    marginTop: 0,
    position: 'absolute',
  },
  background:{
    flex: 3,
    marginBottom: 'auto',
    backgroundColor: Colors.SANTA_GRAY,
  },
  circle:{
    position: 'absolute',
    marginTop: -50,
    marginLeft: 20,
    width: 100,
    height: 110,
    borderRadius: 100 / 2,
    backgroundColor: "lime",
    overflow: 'hidden',
  },
  button:{
    backgroundColor: Colors.SCOTCH_MIST_TAN,
    textAlign: 'center',
    padding: 30,
    fontSize: 38,
    marginTop: 60,
    marginHorizontal: 20,
    borderColor: '#20232a',
    borderWidth: 1,
    borderRadius: 8,
  },
  buttonDark:{
    backgroundColor: Colors.LICHEN,
    textAlign: 'center',
    padding: 30,
    fontSize: 38,
    marginTop: 60,
    marginHorizontal: 20,
    borderColor: '#20232a',
    borderWidth: 1,
    borderRadius: 8,
  },
  buttonText:{
    textAlign: 'center',
    fontSize: 32,
    fontFamily: 'Domine-Regular'
  },
  whitebox: {
    backgroundColor: 'white',
    marginTop: -40
  },
  pressable: {
    marginTop: 70
  }
});

