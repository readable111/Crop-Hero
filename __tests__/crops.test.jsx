import { render, fireEvent } from '@testing-library/react-native'
import { useRouter } from 'expo-router'
import crops from '../app/crops'
import React from 'react'
import AsyncStorage from '@react-native-async-storage/async-storage'
import * as Font from 'expo-font'

jest.mock('expo-router', () => ({
    router:{
        push: jest.fn(),
    },
}
))

jest.mock('expo-font', () => ({
    useFonts: jest.fn(),
    loadAsync: jest.fn().mockResolvedValue(true),
    isLoaded: jest.fn(()=>true)
}))

describe('<crops/>', () =>{
    beforeEach(async () => {
        await AsyncStorage.setItem("dark_mode_setting", false);

        Font.useFonts.mockReturnValue([true, false])
        
        const [fontsLoaded, fontError] = Font.useFonts()
        expect(fontsLoaded).toBe(true)
        expect(fontError).toBe(false)

    })

    it('should go to addCrops page on button press', () => {
        const { getByTestId } = render(
            <crops/>
        )
        
        const addButton = getByTestId('addCrops');
        fireEvent.press(homeButton);

        expect(require('expo-router').router.push).toHaveBeenCalledWith('/home');
    })

    test('renders correctly', () =>{
        const tree = render(<crops/>).toJSON();
        expect(tree).toMatchSnapshot();
    })

    

})