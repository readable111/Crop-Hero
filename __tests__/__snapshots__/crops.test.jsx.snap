// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Crops/> renders correctly 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <Text
    style={
      {
        "alignContent": "center",
        "backgroundColor": "#F1DDBF",
        "borderColor": "#20232a",
        "borderWidth": 1,
        "flex": 0.3,
        "fontFamily": "Domine-Medium",
        "fontSize": 42,
        "padding": 10,
        "textAlign": "right",
      }
    }
  >
    My Crops
  </Text>
  <View
    style={
      {
        "backgroundColor": "#97A5BF",
        "flex": 3,
        "marginBottom": "auto",
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#FFFADA",
          "borderColor": "#20232a",
          "borderRadius": 8,
          "borderWidth": 1,
          "fontSize": 38,
          "marginHorizontal": 20,
          "marginTop": 60,
          "padding": 30,
          "textAlign": "center",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        testID="addCrops"
      >
        <Text
          style={
            {
              "fontFamily": "Domine-Regular",
              "fontSize": 32,
              "textAlign": "center",
            }
          }
        >
          Add Crop
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "backgroundColor": "#FFFADA",
          "borderColor": "#20232a",
          "borderRadius": 8,
          "borderWidth": 1,
          "fontSize": 38,
          "marginHorizontal": 20,
          "marginTop": 60,
          "padding": 30,
          "textAlign": "center",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <Text
          style={
            {
              "fontFamily": "Domine-Regular",
              "fontSize": 32,
              "textAlign": "center",
            }
          }
        >
          View Crop
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "height": 130,
          "marginTop": "auto",
          "width": "100%",
        }
      }
      testID="navbar-light"
    >
      <View
        style={
          {
            "backgroundColor": "clear",
            "flex": 20,
            "height": "auto",
            "marginHorizontal": "auto",
            "width": "100%",
          }
        }
      >
        <View
          style={
            [
              {
                "flexDirection": "row",
              },
              null,
              {
                "height": 50,
              },
            ]
          }
        >
          <View
            style={
              [
                {
                  "flex": 4,
                },
                {
                  "justifyContent": "center",
                  "paddingLeft": 5,
                },
                {
                  "alignItems": "center",
                },
              ]
            }
          />
          <View
            style={
              [
                {
                  "flex": 4,
                },
                {
                  "justifyContent": "center",
                  "paddingLeft": 5,
                },
                {
                  "alignItems": "center",
                },
              ]
            }
          />
          <View
            style={
              [
                {
                  "flex": 4,
                },
                {
                  "justifyContent": "center",
                  "paddingLeft": 5,
                },
                {
                  "alignItems": "center",
                },
              ]
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "#F1DDBF",
                  "borderRadius": 40,
                  "height": 80,
                  "justifyContent": "center",
                  "marginBottom": -83,
                  "width": 80,
                }
              }
            />
          </View>
          <View
            style={
              [
                {
                  "flex": 4,
                },
                {
                  "justifyContent": "center",
                  "paddingLeft": 5,
                },
                {
                  "alignItems": "center",
                },
              ]
            }
          />
          <View
            style={
              [
                {
                  "flex": 4,
                },
                {
                  "justifyContent": "center",
                  "paddingLeft": 5,
                },
                {
                  "alignItems": "center",
                },
              ]
            }
          />
        </View>
      </View>
      <View
        style={
          {
            "backgroundColor": "#F1DDBF",
            "flex": 20,
            "height": 100,
            "marginHorizontal": "auto",
            "width": "100%",
          }
        }
      >
        <View
          style={
            [
              {
                "flexDirection": "row",
              },
              null,
              {
                "height": 35,
              },
            ]
          }
        >
          <View
            style={
              [
                {
                  "flex": 4,
                },
                {
                  "justifyContent": "center",
                  "paddingLeft": 5,
                },
                {
                  "alignItems": "center",
                },
              ]
            }
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "opacity": 1,
                }
              }
              testID="crops"
            >
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  [
                    {
                      "color": "#00DC40",
                      "fontSize": 35,
                    },
                    undefined,
                    {
                      "fontFamily": "material-community",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    {},
                  ]
                }
              >
                Û±îê
              </Text>
            </View>
          </View>
          <View
            style={
              [
                {
                  "flex": 4,
                },
                {
                  "justifyContent": "center",
                  "paddingLeft": 5,
                },
                {
                  "alignItems": "center",
                },
              ]
            }
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "opacity": 1,
                }
              }
              testID="todo"
            >
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  [
                    {
                      "color": "#78938A",
                      "fontSize": 30,
                    },
                    undefined,
                    {
                      "fontFamily": "material-community",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    {},
                  ]
                }
              >
                Û±ì©
              </Text>
            </View>
          </View>
          <View
            style={
              [
                {
                  "flex": 4,
                },
                {
                  "justifyContent": "center",
                  "paddingLeft": 5,
                },
                {
                  "alignItems": "center",
                },
              ]
            }
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "#00AF33",
                  "borderRadius": 27.5,
                  "height": 55,
                  "justifyContent": "center",
                  "marginTop": -30,
                  "opacity": 1,
                  "width": 55,
                }
              }
              testID="home-button"
            >
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  [
                    {
                      "color": "#6FCC8A",
                      "fontSize": 35,
                    },
                    undefined,
                    {
                      "fontFamily": "material-community",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    {},
                  ]
                }
              >
                Û∞ö°
              </Text>
            </View>
          </View>
          <View
            style={
              [
                {
                  "flex": 4,
                },
                {
                  "justifyContent": "center",
                  "paddingLeft": 5,
                },
                {
                  "alignItems": "center",
                },
              ]
            }
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "opacity": 1,
                }
              }
              testID="datahub"
            >
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  [
                    {
                      "color": "#78938A",
                      "fontSize": 35,
                    },
                    undefined,
                    {
                      "fontFamily": "material-community",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    {},
                  ]
                }
              >
                Û±ïé
              </Text>
            </View>
          </View>
          <View
            style={
              [
                {
                  "flex": 4,
                },
                {
                  "justifyContent": "center",
                  "paddingLeft": 5,
                },
                {
                  "alignItems": "center",
                },
              ]
            }
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "opacity": 1,
                }
              }
              testID="profile"
            >
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  [
                    {
                      "color": "#78938A",
                      "fontSize": 35,
                    },
                    undefined,
                    {
                      "fontFamily": "material-community",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    {},
                  ]
                }
              >
                Û∞Äì
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            [
              {
                "flexDirection": "row",
              },
              null,
              {
                "height": 25,
              },
            ]
          }
        >
          <View
            style={
              [
                {
                  "flex": 4,
                },
                {
                  "justifyContent": "center",
                  "paddingLeft": 5,
                },
                {
                  "alignItems": "center",
                },
              ]
            }
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "opacity": 1,
                }
              }
            >
              <Text
                style={
                  {
                    "color": "#00DC40",
                    "fontFamily": "Domine-Regular",
                    "fontSize": 14,
                    "marginTop": 0,
                  }
                }
              >
                Crops
              </Text>
            </View>
          </View>
          <View
            style={
              [
                {
                  "flex": 4,
                },
                {
                  "justifyContent": "center",
                  "paddingLeft": 5,
                },
                {
                  "alignItems": "center",
                },
              ]
            }
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "opacity": 1,
                }
              }
            >
              <Text
                style={
                  {
                    "color": "#78938A",
                    "fontFamily": "Domine-Regular",
                    "fontSize": 14,
                    "marginTop": 0,
                  }
                }
              >
                Notebook
              </Text>
            </View>
          </View>
          <View
            style={
              [
                {
                  "flex": 4,
                },
                {
                  "justifyContent": "center",
                  "paddingLeft": 5,
                },
                {
                  "alignItems": "center",
                },
              ]
            }
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "opacity": 1,
                }
              }
            >
              <Text
                style={
                  {
                    "color": "#78938A",
                    "fontFamily": "Domine-Regular",
                    "fontSize": 14,
                    "marginTop": 0,
                  }
                }
              >
                Home
              </Text>
            </View>
          </View>
          <View
            style={
              [
                {
                  "flex": 4,
                },
                {
                  "justifyContent": "center",
                  "paddingLeft": 5,
                },
                {
                  "alignItems": "center",
                },
              ]
            }
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "opacity": 1,
                }
              }
            >
              <Text
                style={
                  {
                    "color": "#78938A",
                    "fontFamily": "Domine-Regular",
                    "fontSize": 14,
                    "marginTop": 0,
                  }
                }
              >
                Data Hub
              </Text>
            </View>
          </View>
          <View
            style={
              [
                {
                  "flex": 4,
                },
                {
                  "justifyContent": "center",
                  "paddingLeft": 5,
                },
                {
                  "alignItems": "center",
                },
              ]
            }
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "opacity": 1,
                }
              }
            >
              <Text
                style={
                  {
                    "color": "#78938A",
                    "fontFamily": "Domine-Regular",
                    "fontSize": 14,
                    "marginTop": 0,
                  }
                }
              >
                Profile
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
